# Introducción


Se denomina ruido a toda señal no deseada que se mezcla con la señal útil que se quiere recibir. El ruido se debe a múltiples causas  como al ruido térmico de los resistores, a las interferencias de señales externas, etc.  No es posible eliminar totalmente el ruido, ya que los componentes electrónicos no son perfectos y por la temperatura a la que operen. Sin embargo, es posible minimizar su efecto.

La distorsión que produce el ruido en una determinada comunicación depende de su potencia, de su distribución espectral respecto al ancho de banda de la señal y de la propia naturaleza de información que transporta. 
Como el ruido es un proceso aleatorio, puede ser caracterizado por variables como varianza, distribución y destino espectral.

La distribución espectral del ruido puede variar por la frecuencia, y su densidad de potencial es medida en  $\left[ \frac{W}{Hz}\right]$.
Como la potencia de un elemento resistivo es proporcional a la raíz cuadrada del voltaje alrededor del elemento, la unidad de la densidad de voltaje del ruido podría escribirse como $\left[ \frac{V}{\sqrt{Hz}}\right]$.
Se define como ruido blanco aquel que su densidad espectral de ruido en frecuencia es constante. 
Para medir la influencia del ruido sobre la señal se utiliza la relación señal/ruido, que generalmente se emplea en $dB$. 



## Tipos de ruido

### Ruidos correlacionados
El ruido correlacionado es producido por amplificaciones no lineales e incluye armónicos y distorsión de intermodulación, que son dos formas de distorsión no lineal.

Distorsión armónica: Se define como la producción de armónicos de una señal originados por una mezcla no lineal. Los armónicos son múltiplos enteros de la señal original de entrada, la señal original es la primer armónica y se conoce como la frecuencia fundamental.

Distorsión de intermodulación: se refiere a la generación indeseable de productos cruzados que son la suma o restas de frecuencias. 

### Ruidos no correlacionados

Los ruidos no correlacionados pueden tener su origen en fuentes externas: como ruidos de conmutación (fuentes de alimentación), ruidos de origen atmosférico (como rayos producidos por tormentas o la lluvia de partículas  producida por rayos cósmicos con su interacción con la atmósfera) o puede tener origen en el propio tratamiento de la señal por el circuito electrónico.  
El ruido en circuitos electrónicos provienen de distintas fuentes. Vamos a destacar tres fuentes de ruido: Thermal Noise, Shot Noise y Flicker Noise.

### Ruido de disparo

El ruido de disparo en dispositivos electrónicos es el resultado de fluctuaciones estadísticas aleatorias inevitables de la corriente eléctrica cuando los portadores de carga (como los electrones) atraviesan un espacio. Si los electrones fluyen a través de una barrera, entonces tienen tiempos discretos de llegada. Esas llegadas discretas exhiben ruido de disparo. Típicamente, se usa la barrera en un diodo. [3] El ruido de disparo es similar al ruido creado por la lluvia que cae sobre un techo de hojalata. El flujo de lluvia puede ser relativamente constante, pero las gotas de lluvia individuales llegan discretamente.

El valor de la raíz cuadrada media de la corriente de ruido de disparo está dada por la fórmula de Schottky,

$$i_n = \sqrt[]{2 I q \Delta B}$$

donde $I$ es la corriente continua, $q$ es la carga de un electrón y $\Delta B$ es el ancho de banda en hertz. La fórmula de Schottky supone llegadas independientes.

Los tubos de vacío presentan ruido de disparo porque los electrones abandonan aleatoriamente el cátodo y llegan al ánodo (placa). Es posible que un tubo no muestre el efecto de ruido de disparo completo: la presencia de una carga espacial tiende a suavizar los tiempos de llegada (y así reducir la aleatoriedad de la corriente).

Los conductores y las resistencias normalmente no muestran ruido de disparo porque los electrones se encuentran en equilibrio térmico y se mueven difusamente dentro del material; los electrones no tienen tiempos de llegada discretos. El ruido de disparo se ha demostrado en resistencias mesoscópicas cuando el tamaño del elemento resistivo se vuelve más corto que la longitud de dispersión del electrón-fonón. 

Debe distinguirse el ruido de disparo de las fluctuaciones de corriente en equilibrio, las cuales se producen sin aplicar ningún voltaje y sin necesidad de que exista ningún flujo promedio de corriente. Estas fluctuaciones de la corriente de equilibrio se conocen como ruido de Johnson-Nyquist.

El ruido de disparo se puede modelar como un proceso de Poisson y los portadores de carga que forman la corriente siguen una distribución de Poisson. El ruido de disparo está yuxtapuesto a cualquier ruido presente, y se puede demostrar que es aditivo respecto al ruido térmico y a él mismo.

### Ruido Flicker

Ruido Flicker o ruido $1/f$ presenta un densidad espectral que crece, por debajo de kilo-hertz, al disminuir la frecuencia. 
También es llamado ruido en exceso o ruido de semiconductor, habiéndose atribuido diferentes orígenes, entre ellos los procesos aleatorios de generación-recombinación térmica de pares electrón-hueco, dicho en otras palabras es una señal con una frecuencia de espectro que cae, proporcional a la magnitud de la señal, es más importante a bajas frecuencias y para anchos de banda pequeños $1/f$es aproximadamente constante.

### Ruido Térmico

El ruido térmico (a veces térmico, Johnson o Nyquist) es inevitable y se genera por el movimiento térmico aleatorio de los portadores de carga (generalmente electrones), dentro de un conductor eléctrico, que ocurre independientemente de cualquier voltaje aplicado. Este ruido sera desarrollado en la siguiente sección.


# Ruido Térmico en Resistores.


También conocido como ruido Johnson-Nyquist), es el ruido generado por el equilibrio de las fluctuaciones de la corriente eléctrica dentro de un conductor eléctrico, el cual tiene lugar bajo cualquier voltaje, debido al movimiento térmico aleatorio de los electrones.

El ruido térmico es aproximadamente blanco, lo que significa que su densidad espectral de potencia es casi plana (hasta frecuencias cercanas a los THz). Además, la amplitud de la señal sigue una distribución gaussiana. 


Como Nyquist afirmó en su artículo de 1928, la suma de la energía en los modos normales de oscilación eléctrica determinaría la amplitud del ruido. 
Nyquist utilizó el teorema de Equipartición de Boltzmann y Maxwell. 

La constante de Boltzmann $k_{\rm B}$  es la constante física que relaciona temperatura absoluta $T$ en $[K]$ y energía.

La densidad de potencia del ruido en $[W/Hz]$ se puede calcular como:

$$\left \langle H \right \rangle =k_{\rm B} T$$

La constante de Boltzman toma un valor aproximado de:

$$k_B \; \approx \; 1,38064852 \times 10^{-23}  J/K$$ 


Multiplicando la ecuación por el ancho de banda se obtiene el resultado como potencia de ruido.

$$N=k_{\rm B} T \Delta f  \frac{\frac{h f}{k_{\rm B} T}}{e^{\frac{hf}{k_{\rm B} T}-1}}$$ 

donde $N$ es la potencia de ruido, $\Delta f$ es el ancho de banda y $h$ = Constante de Planck
 

Es posible simplificar esta ecuación, donde es puede usarse hasta $100\,GHz$ y temperaturas mayores a los $100\,K$:

$$N=k_{\rm B} T \Delta f$$ 


## Fuente de tensión de ruido blanco 

El ruido térmico es diferente del  ruido de disparo, que tiene lugar cuando el número finito de  electrones es suficientemente pequeño para dar lugar a la aparición de fluctuaciones estadísticas apreciables en una medición. 

La definición de ruido de Johnson-Nyquist aplica a cualquier tipo de medio conductor. Puede modelarse como una fuente de tensión que representa el ruido de una  resistencia eléctrica no ideal en serie con una  resistencia eléctrica libre de ruido. 

Los modelos equivalentes de un resistor se muestran en la siguiente figura.

![1-cuatrocientos](fig/noiseVolt.png)

La densidad espectral de potencia de ruido en un resistor esta dada por:

$$\bar{v_{n}^2} =  4 k_B T R \,\,\,[\mathrm{V^2}/\mathrm{Hz}]$$


Donde $k_B$ es la  constante de Boltzmann, $T$  es la temperatura a la que se halla el resistor en grados kelvin [$K$], y $ R $ su valor en Ohmios [$\Omega$].


Observar que sucedería si fuera posible medir el valor de tensión con un instrumento de ancho de banda infinito, el valor de tensión medido seria infinito. Esto no es posible debido a que como mencionamos anteriormente, la aproximación de densidad espectral de potencia plana solo es valida hasta frecuencias cercanas a los THz, para frecuencias más altas, se debe tener en cuenta que la densidad espectral toma la siguiente forma.


$$N_{Res}(f) = 4 k_B T R B \frac{\frac{h f}{k_B T}}{e^{\frac{hf}{k_{\rm B} T}-1}}$$

La expresión anterior es similar a la expresión de la irradiancia espectral de un cuerpo negro y es así debido a la estrecha relación entre la agitación térmica de los electrones dentro de un solido y la radiación electromagnética que emite.


Para frecuencias bajas $hf << h f$ por lo tanto, la densidad espectral se reduce a la expresión $N_{Res} = 4 k_B T R$.

Si queremos encontrar la potencia de ruido generada en un ancho de banda B, solo debemos multiplicar la densidad espectral de potencia por el mismo. 

Es posible simplificar esta ecuación, donde es puede usarse hasta $100\,GHz$ y temperaturas mayores a los $100\,K$:

Por lo tanto, el valor cuadrático medio de la tensión en el resistor esta dado por por:

$${\bar{v_{n}^2}} =4 k_B T R B$$

Donde  

$B$ = ancho de banda en Hz 

$R$ = resistencia  
 


## Fuente de corriente de ruido blanco 

Utilizando el teorema de Norton es posible obtener el dual de la expresión anterior para la corriente de ruido en el resistor.


$${\bar{i_{n}^2}} = {4 k_B T G B}$$


Donde G es la inversa de la resistencia R, su conductancia.


![2-cuatrocientos](fig/noiseAmpere.png)



##  Simulación de la densidad de ruido espectral.

Cada resistencia ofrece en sus terminales una Potencia de ruido disponible de:

$$k_B T B = (kT) \cdot B = (Densidad\, de\,  potencia\, de\, ruido) \times (Ancho\, de\, banda)$$

La densidad de potencia de ruido es la potencia que se puede medir en cada Hz del ancho de banda. El total potencia disponible se obtiene si se multiplica esta densidad por el ancho de banda válido (o por integración sobre el ancho de banda si la densidad de potencia de ruido varía con el ancho de banda).


Pero frecuentemente se prefiere manejar voltaje en lugar de la potencia, porque es más fácil medir voltajes a diferentes frecuencias que la potencia. Entonces se usa la ecuación de potencia para operar con los dos  valores:

$$P = \frac{V^2}{R}$$

Cuando se calcula la raíz cuadrada de la Densidad de potencia de ruido, obtendrá una expresión para la "Densidad de voltaje de ruido espectral", dada en [V/$\sqrt[]{Hz}$]. 
Si se multiplica este valor por $\sqrt[]{Hz}$, se obtiene el voltaje total de ruido. 


### Ejemplo 1

Para una resistencia de $1 K\Omega$  a $ 300 K$ el valor medio cuadrático de ruido de la tensión sobre el resistor es: 


$$\sqrt{\bar{v_{n}^2}}= \sqrt{4 \cdot 1.38\times10^{-23} \frac{J}{K} \cdot 300\,\mathrm{K} \cdot 1~\mathrm{k}\Omega \cdot  } = 4.07  ~\mathrm{  \frac{\mu V}{\sqrt{Hz}}}$$ 

Valor de tensión sobre el resistor: $4.07e-09 V/\sqrt{Hz}$


In [4]:
# Para los calculos
import numpy as np


import scipy                  # http://scipy.org/
from scipy import constants

import matplotlib.pyplot as plt
from IPython.display import Image  
%matplotlib inline
params = {'legend.fontsize': 24,
          'figure.figsize': (12, 8),
         'axes.labelsize': 24,
         'axes.titlesize': 24,
         'xtick.labelsize':24,
         'ytick.labelsize':24}
plt.rcParams.update(params) 




In [2]:
# Calculo con Jupyter
kb = constants.value('Boltzmann constant')
T  = 300
R  = 1e3

# densidad espectral de tension de ruido
v2  = (4*kb*T*R)**(1/2)  # [V/sqrt(Hz)]

# presenta los resultados 
print('Valor de tensión sobre el resistor: {:1.2e} V/(Hz)^(1/2)'.format(v2) ) 


Valor de tensión sobre el resistor: 4.07e-09 V/(Hz)^(1/2)


## Simulación con LTspice, midiendo la densidad de tensión de ruido 

![3-cuatrocientos](fig/resistencia.png)


## Asociación de resistores en serie:

Considérese 2 resistores, cada uno con su modelo equivalente de ruido


![3-cuatrocientos](fig/resis_serie.png)

La suma de los resistores es $R_1$ + $R_2$ y su equivalente de ruido es:

![3-cuatrocientos](fig/resis_serie_fin.png)

Se suman los cuadrados de los valores cuadráticos medios de los generadores de ruido. 


## Resistores en paralelo

Cuando los resistores se hallan en paralelo suele usarse el siguiente esquema:

![3-cuatrocientos](fig/resis_para_1.png)

![3-cuatrocientos](fig/resis_para_2.png)


Asociaciones mixtas
Supóngase un circuito resistivo como el siguiente:

![3-cuatrocientos](fig/resis_mix_1.png)

Supóngase querer conocer la tensión de ruido en Ri si esta última no tiene ruido, se puede hacer un Thevenin quedando:


# Cifra de ruido y Factor de ruido:

La cifra de ruido (NF) y el factor de ruido (F) son medidas de degradación de la relación señal/ruido (SNR) causada por componentes en una cadena de señal. 

El factor de ruido es, por lo tanto, la relación entre el ruido de salida real y el que se mantendría si el propio dispositivo no introdujera ruido, o la relación de la entrada SNR a la salida SNR.

$$F  =  \frac{\frac{S_i}{N_i}}{\frac{S_o}{N_o}}$$

La cifra de ruido es simplemente el factor de ruido (F) expresado en decibelios (dB).

$$NF = 10 \cdot log_{10}(\frac{SNR_i}{SNR_o})$$ 


Operando el factor de ruido, es posible obtener otras formas de calcularlo.

$$F = \frac{ S_i \cdot N_o}{ S_o \cdot  N_i}$$

$$F = \frac{ S_i \cdot N_o}{ G \cdot S_i \cdot N_i}$$

La potencia de ruido a la salida $N_o$ puede ser expresar como $N_o = G N_o + N_e$, entonces:   

$$F = \frac{  G \cdot N_i + N_e }{ G \cdot N_i}$$

$$F =  1 + \frac{ N_e }{ G \cdot N_i}$$

$N_e$ es la potencia en exceso a la salida y  se puede calcular como:

$$N_{ei} =  (F - 1) \cdot  G \cdot N_i$$

Para un circuito paralelo (modelo admitancia), la potencia adicional puede ser suministrada por una fuente de corriente.

$$i_{ei}^2 \cdot Rto =  (F - 1) \cdot  G \cdot i_i^2 \cdot Rti$$

$$i_{ei} =  \sqrt{(F - 1) \frac{Rti}{Rto} \dot G } \cdot   i_i$$

donde $i_i = \sqrt{4\cdot k_b \cdot T \cdot B \cdot \frac{1}{R} }$



Si la fuente $N_e$ esta en la entrada, $N_e = G \cdot N_ei$

$$F = 1 + \frac{G \cdot N_{ei} }{ G \cdot N_i}$$

$$F =  1 + \frac{ N_{ei} }{ N_i}$$

La potencia $N_{ei} $ corresponde a una potencia de ruido excedente que se presenta en la entrada y  se puede calcular como:

$$N_{ei} =  (F - 1) \cdot   N_i $$


Para un circuito paralelo (modelo admitancia), la potencia adicional puede ser suministrada por una fuente de corriente.

$$i_{ei}^2 \cdot Rti =  (F - 1) \cdot   i_i^2 \cdot Rti$$

$$i_{ei} =  \sqrt{(F - 1) } \cdot   i_i$$

donde $i_i = \sqrt{4\cdot k_b \cdot T \cdot B \cdot \frac{1}{R} }$




## Fuentes de ruido en LTspice

No parece haber una forma directa de agregar un ruido de voltaje o una fuente de ruido de corriente a un circuito LTspice (u otro tipo de Spice) para ser usado en una simulación de ruido. 
Sin embargo, es posible agregar fuentes de ruido para usar en simulaciones .tran (dominio del tiempo) utilizando fuentes de comportamiento.  
Un método para agregar ruido blanco (Johnson y disparo) y $1/f$ (flicker) o fuentes de ruido de corriente de la amplitud deseada para usar en simulaciones de ruido (.noise).

Un caso en el que dichas fuentes de ruido pueden ser útiles es cuando se realizan modelos de simulación de amplificadores  donde se conoce el voltaje de entrada referido y los ruidos de corriente a partir de la hoja de datos.
La única fuente de ruido simple (que afecta a las simulaciones de ruido) en LTspice es una resistencia simple. Existen otras fuentes de ruido en los modelos de dispositivos semiconductores, pero esos modelos son más complejos. 
Una resistencia ideal tiene un ruido de voltaje descrito por:

$$Vrms  = \sqrt{4 \cdot k_B \cdot T \cdot R \cdot B }$$

donde $k_B$ es la constante de Botzmann ($1.381\times 10^{-23} J/K$), $T$ es la temperatura en Kelvin ($300\, K$ por defecto en LTspice), $B$ es el ancho de banda en [Hz] y $R$ es La resistencia en [$\Omega$].


En las hojas de datos de los transistores o amplificadores generalmente especifica el ruido de voltaje blanco en unidades de $[\frac{nV}{\sqrt{Hz}}]$ y el ruido de corriente en $[\frac{pA}{\sqrt{Hz}}]$ (a veces $[\frac{fA}{\sqrt{Hz}}]$).

Para crear una fuente de ruido de voltaje blanco, podemos conectar los terminales de entrada de una fuente de voltaje dependiente de voltaje a una resistencia y usar un factor de escala adecuado. 
La fuente dependiente aísla la resistencia de cualquier circuito que esté conectado a ella y preserva la amplitud de ruido de voltaje independientemente de la carga.


Dado que las fuentes de ruido generalmente se especifican en $nV/\sqrt{Hz}$,  es conveniente poder ingresar ese número directamente como parte del modelo. 
Una forma simple de hacerlo es seleccionando una resistencia que produce una densidad de ruido de  $1 \frac{nV}{\sqrt{Hz}}$ e ingrese la amplitud de ruido de la hoja de datos como la ganancia de voltaje de la fuente dependiente. 


Resolviendo la ecuación anterior para $v_n = 1 nV$ cuando $T = 300 K$ y $B = 1 Hz$ da $R = 60.342$.

El esquema de la especificación LTspice resultante para una fuente de ruido de voltaje de 2 $\frac{nV}{\sqrt{Hz}}$ es así:

![4-cuatrocientos](fig/voltnoiseLT.png)



De manera similar, para crear una fuente de ruido de corriente blanca, podemos usar una fuente de corriente dependiente de voltaje ( Fuente de G). 
Para establecer el factor de transconductancia de la fuente a la densidad de ruido en $[\frac{pA}{\sqrt{Hz}}]$, es necesaria una resistencia con una densidad de ruido de 1 $\frac{pA}{\sqrt{Hz}}$, lo que significa que la resistencia tendrá un valor de $60.342e-6 $. 
 
Una fuente de ruido de corriente con una densidad de ruido de 2 $\frac{pA}{\sqrt{Hz}}$ se puede modelar así:


![5-cuatrocientos](fig/currentnoiseLT.png)



Para integrar el ruido en un ancho de banda seleccionado en un análisis de ruido, se debe presionar Ctrl y hacer clic en la etiqueta de seguimiento de datos, V(ruido), en el visor de formas de onda; se muestra el ruido RMS total basado en el ancho de banda especificado en la directiva .noise, sobre los limites graficados. 

Si prefiere calcular el ruido total en un ancho de banda limitado en lugar del ancho de banda especificado en la directiva .noise, puede modificar los límites de frecuencia inferior y superior del visor de formas de onda haciendo clic en el eje horizontal. 
Una vez que tenga su forma de onda mostrando el ancho de banda de interés, presionar Ctrl y hacer clic en la etiqueta de seguimiento de datos para mostrar el ruido RMS total basado en el ancho de banda de interés.

Alternativamente, puede agregar la directiva .measure a su esquema que calculará el ruido total en un ancho de banda particular y verá sus resultados en el registro de errores de LTspice (Ctrl + L) después de cada simulación. 

.measure noise NOISERMS integ V(onoise) from 1e6 to 1e9

Puede encontrar más información sobre el uso de la instrucción .measure en la Ayuda de LTspice (F1).

In [12]:
# Ruido en la entra dado las resistencia de entrada
T  = 300
B  = 1
kb = constants.value('Boltzmann constant')

vn = 100e-12

rn = vn**2/(4*kb*T)

print('.param rn = {:1.3e}'.format(rn))



.param rn = 6.036e-01


Ejemplo 2


![5-cuatrocientos](fig/eje2.png)




Calcular el valor cuadrático medio de la tensión sobre el resistor $R_1$, siendo $R_g =50\Omega$ , $R_1 =100\Omega$, $\bar{v_{g}}^2= (100 pV)^2$, $T=300 K$  y un ancho de banda $B= 1 MHz$. Compare los resultados para un ancho de banda $B = 10 MHz$.


In [44]:
# Ruido en la entra dado las resistencia de entrada
T  = 300
B  = 1
kb = constants.value('Boltzmann constant')

vn = 100e-12

rg = 50

R1 = 100

rn = vn**2/(4*kb*T)

print('.param rn = {:1.3e}'.format(rn))


von = vn**2*(R1/(rg+R1))**2


vg2  = (4*kb*T*rg)
vog = vg2*(R1/(rg+R1))**2

v12  = (4*kb*T*R1)
vo1 = v12*(rg/(rg+R1))**2

vo = von + vog + vo1

print('vo1**2 = {:3.1e}V**2'.format(vo1**(1)))
print('vog**2 = {:3.1e}V**2'.format(vog**(1)))
print('von**2 = {:3.1e}V**2'.format(von**(1)))


.param rn = 6.036e-01
vo1**2 = 1.8e-19V**2
vog**2 = 3.7e-19V**2
von**2 = 4.4e-21V**2


Potencia de Ruido Disponible

Volvamos al ejercicio 2, pero supongamos que $R_1$ no es capaz de generar ruido.

![5-cuatrocientos](fig/eje2.png)


Calculemos la potencia de ruido en $R_1$, para ello planteemos el valor cuadrático medio de la tensión en $R_1$ debido al ruido generado por $R_g$. EL valor de tensión esta dado por la siguiente expresión


$$\bar{v_{RL}^2}= 4k_BTR_gB*(\frac{R_1}{R_g+R_1})^2 \,\,\,[V^2]$$

La cual es máxima cuando $R_1=R_g$ y toma el valor

$$\bar{v_{R1}^2}= k_BTR_1B \,\,\,[V^2]$$

Para obtener la potencia de ruido disipada en $R_1$ solo debemos dividir la tensión cuadrática media en $R_1$ por el valor de $R_1$. Por lo tanto la potencia disponible de ruido en el resistor es 


$$P_{Disp}= k_BTB \,\,\,[V^2/\Omega]= [W]$$

Como se ve, la potencia disponible es independiente del valor del resistor. Para obtener la densidad espectral de potencia disponible solo debemos dividir por el ancho de banda $B$.

$$N_{Disp}= k_BT = [W/Hz]$$


Por lo tanto la máxima potencia de ruido que se puede transmitirse cuando los circuitos están adaptados (Para máxima transferencia de energía) depende del ancho de banda y de la temperatura física a la que se encuentra resistor.

Ejemplo 

Calcule la densidad espectral de ruido y la  potencia de ruido en $dBm/Hz$ y $dBm$ respectivamente,  que desarrolla un resistor a una temperatura de $T=290K$ sobre un ancho de banda $B$.
Aplicando las fórmulas anteriores y  expresándolas en dBm se tiene

$$10*\log_{10}{(\frac{N_{Disp}}{1mW})}= 10*\log_{10}{(\frac{k_B T}{1mW})} = -174 \,\,\,[dBm/Hz]$$

$$10*\log_{10}{(\frac{P_{Disp}}{1mW})}= 10*\log_{10}{(\frac{k_B T B}{1mW})} = {-174 + 10*\log_{10}{(B)}} \,\,\,[dBm]$$

Observar que existe un nivel limite inferior para el ruido y todo lo que este por debajo del mismo no podrá ser distinguido, por lo tanto debemos procurar estar siempre trabajando con potencias muy superiores al nivel "piso" de ruido.


## Temperatura equivalente de ruido

Como vimos anteriormente, la potencia de ruido es proporcional a la temperatura física del resistor.Ahora supongamos un dispositivo cualquiera el cual no sabemos de que esta compuesto pero genera una potencia de ruido $P_n$ , sobre un ancho de banda $B_0$. Este dispositivo puede ser modelado como un resistor el cual esta a una temperatura ficticia (no física) pero que genera una potencia de ruido $P_n$. El siguiente diagrama muestra lo mencionado anteriormente.

![5-cuatrocientos](fig/tempEq.png)

Donde $P_n$ es igual a:

$$P_{n}= k_BT_{eq}B_0 \,\,\,[W]$$

Despejando $T_{eq}$ en función de la potencia de ruido y el ancho de banda

$$T_{eq}=\frac{P_n}{k_BB_0} \,\,\,[K]$$
En muchos casos se hablará de temperatura equivalente de ruido como indicador del nivel de ruido generado en un circuito, pero debemos tener en cuenta que no es una temperatura física y muchas veces depende de la temperatura física y de la polarización. 

#### Ejemplo 3

Calcule la temperatura equivalente de ruido de un circuito el cual se mude una potencia de ruido $P_n= 1pW$ para un ancho de banda $B=10MHz$. 
Aplicando las fórmulas anteriores

$$T_{eq}=\frac{1pW}{k_B*10 MHz}= 7243 \,\,\,[K]$$

La temperatura obtenida obviamente no es una temperatura física capaz de ser soportada por un resistor. En muchos casos la generación de ruido térmico va a estar dominada por las perdidas en los circuitos pero para llegar a estas potencias se necesitara de elementos activos que lo amplifiquen y o generen otros tipos de ruido de mas potencia como el que genera la polarización.

## Ruido en Antenas

Una de las aplicaciones de la temperatura equivalente de ruido es para describir el ruido presente en los terminales de una antena. Como sabemos, las antenas mas comunes se modelan con dos resistencias, una es la llamada resistencia de radiación que es sobre la que se induce la señal captada por la antena y otra resistencia llamada ohmica que esta relacionada con las pérdidas en los conductores que la forman. El ruido presente en los bornes de la antena esta generado por dos causas, uno es el ruido térmico generado por la resistencia de perdidas, el cual es despreciable, y otro que es ruido captado por la antena y que aparece como ruido generado por la resistencia de radiación. Si bien este ruido en muchos casos no es de naturaleza térmica, puede modelarse como ruido térmico generado por un resistor a una temperatura ficticia llamada temperatura de antena, $T_A$. Por lo tanto, una antena es modelada como una resistencia de valor igual a la suma de las resistencias de radiación y pérdidas y a una temperatura $T_A$.El siguiente ejemplo muestra lo explicado anteriormente.


### Ejercicio 4

Una dipolo fue construido para tener una resistencia de radiación $R_{rad}=70 \Omega$, las pérdidas en la misma son despreciables. La misma posee un ancho de banda $B=100KHz$. La misma se instaló en el sitio donde se utilizará y se midió en bornes de la misma una tensión de ruido $0.3\mu V$. Calcule la temperatura de antena. 

$${V_A}^2 = 4 k_BT_A R_{rad} B \,\,\,[V]$$

$$T_{A}=\frac{{V_A}^2}{4 k_B R_{rad} B}= 233 \,\,\,[K]$$

Debemos tener en cuenta que la temperatura de antena depende del sitio donde se ubique y del tipo de antena. En muchos casos la temperatura de antena se debe estimar para evaluar si es posible instalarla en determinado sitio. Algunas de las fuentes de ruido que afectan a las antenas son: CMB (Radiación Cósmica de Microondas), el Sol, ruido atmosférico, Ruido térmico generado por el mismo suelo etc. En resumen, todo cuerpo a una temperatura mayor a $0K$ generara ruido térmico (radiación de cuerpo negro) y se inducirá en la antena en mayor o menor medida según el patrón de radiación que posea. Para más información consulte [antenas Krauss]


## Ruido en Impedancias

Hasta ahora solo hemos tratado el ruido térmico generado en resistores, y vimos que muchas veces depende del ancho de banda con el que se mida. Pensemos por un momento en cual seria el efecto de agregar componentes reactivos a los circuitos resistivos. Como sabemos de teoría de circuitos, los resistores responden igualmente para todas las frecuencias, mientras que los elementos reactivos responden en función de la frecuencia a la que trabajen. Por lo tanto cuando asociemos resistores y elementos reactivos el efecto de los elementos reactivos no sera añadir ruido, peri si limitar la trasferencia del ruido a determinadas frecuencias y aumentarla a otras, en pocas palabras,el ruido generado no sera plano.
Nyquist determinó que la densidad espectral de ruido generado en una impedancia solo depende del valor resistivo de la impedancia (Parte real). El valor de tensión cuadrático medio en bornes de la impedancia se calcula integrando la densidad espectral sobre el ancho de banda de interés. El cálculo de la tensión de ruido tiene la siguiente expresión.

$${\bar{v_{n}^2}} = {4 k_B T \int_{B} R(f) df} $$

La expresión anterior equivale tener infinitas resistencias de  valor $R(f)$ que solo aportan ruido en un diferencial tensión cuadrática media $4 k_B T R(f) df$ centrado en $f$.

Para fijar conocimientos, consideremos un resistor en paralelo con un capacitor como el que se muestra en la siguiente figura

![5-cuatrocientos](fig/noiseImp.png)

El valor de impedancia en bornes se calcula:

$$Z(f) = \frac{R}{1+j\omega R C}$$

Donde

$$Re[Z(f)] = R(f) = \frac{R}{ 1+ (\omega R C)^2 } $$

Por lo tanto, para conocer el valor cuadrático medio de la tensión de ruido a la salida debemos resolver la siguiente integral

$${\bar{v_{n}^2}} = {4 k_B T R \int_{B} \frac{1}{ 1+ (2 \pi f R C)^2} df}$$

Observar que la constante que multiplica a la integral tiene unidades de $V^2/Hz$, por lo tanto, la integral debe tener unidades de $Hz$. Más adelante veremos que esta integral determina el ancho de banda de ruido equivalente y esta dado pros la respuesta en frecuencia del circuito. Para resolver la siguiente integral, debemos conocer el ancho de banda sobre el que queremos medir el ruido, supongamos que poseemos un instrumento de ancho de banda infinito y queremos verificar el nivel de ruido,la integral que debemos resolver es:

$${\bar{v_{n}^2}} = {4 k_B T R \int_{0}^{\infty} \frac{df}{ 1+ (2 \pi f R C)^2} }$$

Recurriendo a una tabla de integrales entonces:

$${\bar{v_{n}^2}} = \frac{K_B T}{C}$$


Sabiendo que para un filtro RC, la frecuencia de corte se halla en $f_{-3db}=\frac{1}{RC}$. La tensión cuadrática media de ruido toma la expresión:

$${\bar{v_{n}^2}} = {4 k_B T R (\frac{\pi}{2} f_{-3db})}$$

Donde el término entre paréntesis es el denominado ancho de banda equivalente de ruido.


## Ejemplo de Factor de ruido en amplificadores

Suponiendo un transistor con cifra de ruido de $3 dB$, calcular la corriente en excedo de la fuente del transistor. 

La temperatura del sistema es de $T = 300K$

Parametros del transistor:

$$r_{11} = 1k\Omega$$ 

$$g_m    = 100 mS$$

$$r_{22} = 10 k \Omega$$

In [39]:
# Ejemplo de calculo de la fuente de corriente en exceso de ruido  

NF = 3 # dB
F  = 10**(NF/10) 

# Empleando solo la parte resistiva del transistor
r11= 1e3
gm = 100e-3
r22= 10e3

# MTE
rg = r11
rl = r22

# Resistencia total entrada
rti= r11*rg/(r11+rg)
# Resistencia total salida
rto= r22*rl/(r22+rl)

# Ganancia de tension
Av = gm * rto

# Ganancia de Potencia
G = Av**2 * (r11/rl)

# Ruido en la entra dado las resistencia de entrada
T  = 300
B  = 1
kb = constants.value('Boltzmann constant')

vi = (4*kb*T*B*rti)**(1/2)
print('vi = {:3.1f}pV'.format(vi/1e-12))

Ni = vi**2/r11
print('vi = {:3.1f}pV'.format(vi/1e-12))

vo =  (gm*vi*rto)**2 +(4*kb*T*B*rto)

print('vo = {:3.1e}V'.format(vo**(1/2)))

vi = 2878.2pV
vi = 2878.2pV
vo = 1.4e-06V


In [45]:
#####################################
### Fuente de corriente de ruido en exceso en la entrada
#####################################
ii   = (4*kb*T*B/rti)**(1/2)
# si la fuente de ruido esta en la entrada la corriente de ruido ie
iei = (F - 1)**(1/2)  *  ii 

#####################################
### Fuente de corriente de ruido en exceso en la salida
#####################################
# F = (1 + Ne/(G*Ni))
ii   = (4*kb*T*B/rti)**(1/2)
ieo = ((F - 1)*G*rti/rto)**(1/2)*ii   


## Para las simulaciones con LTspice
gei = iei/1e-12
geo = ieo/1e-12

print('.param rg = {:1.3e}'.format(rg))
print('.param rl = {:1.3e}'.format(rl))
print('.param r11 = {:1.3e}'.format(r11))
print('.param r22 = {:1.3e}'.format(r22))
print('.param gm = {:1.3e}'.format(gm))

print('.param gei = {:1.3e}'.format(gei))
print('.param geo = {:1.3e}'.format(geo))


.param rg = 5.000e+01
.param rl = 1.000e+04
.param r11 = 1.000e+03
.param r22 = 1.000e+04
.param gm = 1.000e-01
.param gei = 5.743e+00
.param geo = 2.871e+02


### Medición de la potencia de ruido en la entrada sin ruido en exceso
Se mide la potencia de ruido en la entrada. Esta potencia de ruido es suminstrada por las resistencias del nodo de entrada (generador y transistor).

![6-cuatrocientos](fig/noiseTransistorIn.png)

De la medición, la tensión espectral de ruido en la entrada es:

$V_n =  \sqrt{4\cdot k_b \cdot T \cdot r_{ti} } = 2.87 \frac{n V}{\sqrt{Hz}}$

El valor de tensión rms ($ 1 Hz$ de ancho de banda):

$V_{rms} = 2.87 \frac{n V}{\sqrt{Hz}} \cdot 1 Hz = 2.87 n V$

La potencia de ruido en la entrada del transistor:

$N_{i} =  \frac{ (V_{rms})^2}{r_{11}} = \frac{ (2.87 n V)^2}{1K\Omega} = 8.28\times 10^{-21} W$



### Medición de la potencia de ruido en la salida sin ruido en exceso
![7-cuatrocientos](fig/noiseTransistorOut.png)


De la medición, la tensión espectral de ruido en la entrada es:

$V_n =  \sqrt{4\cdot k_b \cdot T \cdot r_{ti} } = 1.44 \frac{\mu V}{\sqrt{Hz}}$

El valor de tensión rms ($ 1 Hz$ de ancho de banda):

$V_{rms} = 1.44 \frac{\mu V}{\sqrt{Hz}} \cdot 1 HZ = 1.44 \mu V$

La potencia de ruido en la resistencia de salida $r_l$ :

$N_{o} =  \frac{ (V_{rms})^2}{r_{l}} = \frac{ (1.44 \mu V)^2}{10K\Omega} = 2.07\times 10^{-16}  W$

### Ganancia de potencia

$G = \frac{N_o}{N_i} = \frac{2.07\times 10^{-16}}{8.23\times 10^{-21}} = 25.1 \times 10^{3}$

Analiticamente la ganancia de potencia $G$  se puede calcular como:

$G = |Av|^2 \cdot \frac{rl}{r11}  = (gm \cdot r_{to})^2 \cdot \frac{rl}{r11} = 25 \times 10^{3}$


### Medición del Factor de ruido con ruido en exceso en la entrada


![8-cuatrocientos](fig/noiseTransistorInE.png)

De la medición, la tensión espectral de ruido en la salida es:

$V_n =  \sqrt{4\cdot k_b \cdot T \cdot r_{ti} } = 2.03 \frac{\mu V}{\sqrt{Hz}}$

El valor de tensión rms ($ 1 Hz$ de ancho de banda):

$V_{rms} = 2.03 \frac{\mu V}{\sqrt{Hz}} \cdot 1 HZ = 2.03 \mu V$

La potencia de ruido en la resistencia de salida $r_l$ :

$N_{o} =  \frac{ (V_{rms})^2}{r_{l}} = \frac{ (2.03 \mu V)^2}{10K\Omega} = 4.12\times 10^{-16}  W$

El Factor de poptencia:

$F = \frac{N_{o}}{G \cdot N_{i}} = \frac{4.12\times 10^{-16}  W}{ 25\times 10^{3}  \cdot 8.28\times 10^{-21} W} = 2$



### Medición del Factor de ruido con ruido en exceso en la salida
![9-cuatrocientos](fig/noiseTransistorOutE.png)

De la medición, la tensión espectral de ruido en la salida es:

$V_n =  \sqrt{4\cdot k_b \cdot T \cdot r_{ti} } = 2.03 \frac{\mu V}{\sqrt{Hz}}$

El valor de tensión rms ($ 1 Hz$ de ancho de banda):

$V_{rms} = 2.03 \frac{\mu V}{\sqrt{Hz}} \cdot 1 HZ = 2.03 \mu V$

La potencia de ruido en la resistencia de salida $r_l$ :

$N_{o} =  \frac{ (V_{rms})^2}{r_{l}} = \frac{ (2.03 \mu V)^2}{10K\Omega} = 4.12\times 10^{-16}  W$

$F = \frac{N_{o}}{G \cdot N_{i}} = \frac{4.12\times 10^{-16}  W}{ 25\times 10^{3}  \cdot 8.28\times 10^{-21} W} = 2$


#### Nota: en estos calculos se despreciaron los ruidos aportados por las resistencias en la salida.